Sources/SwiftParser/Declarations.swift

```diff
@@ -1369,23 +1369,32 @@ extension Parser {
   }
 
   @_spi(RawSyntax)
-  public mutating func parseFunctionSignature() -> RawFunctionSignatureSyntax {
+  public mutating func parseFunctionSignature(allowOutput: Bool = true) -> RawFunctionSignatureSyntax {
     let input = self.parseParameterClause(for: .functionParameters)
 
     var effectSpecifiers = self.parseDeclEffectSpecifiers()
 
-    let output: RawReturnClauseSyntax?
+    var output: RawReturnClauseSyntax?
 
     if self.at(.arrow) || self.canRecoverTo(TokenSpec(.arrow, allowAtStartOfLine: false)) != nil {
       output = self.parseFunctionReturnClause(effectSpecifiers: &effectSpecifiers, allowNamedOpaqueResultType: true)
     } else {
       output = nil
     }
 
+    var unexpectedAfterOutput: RawUnexpectedNodesSyntax?
+    if !allowOutput,
+      let unexpectedOutput = output
+    {
+      output = nil
+      unexpectedAfterOutput = RawUnexpectedNodesSyntax([unexpectedOutput.raw], arena: self.arena)
```

Comment:
I don’t think you need the `.raw`. If it doesn’t compile without, could you share the error message you are getting?